commits.go

```diff
@@ -0,0 +1,52 @@
+package gogitlab
+
+import (
+	"encoding/json"
+	"time"
+)
+
+const (
+	commit_status = "/projects/:id/repository/commits/:sha/statuses" // Get the statuses of a commit in a project
+)
+
+type CommitStatus struct {
+	Status       string     `json:"status"`
+	CreatedAt    time.Time  `json:"created_at"`
+	StartedAt    *time.Time `json:"started_at"`
+	Name         string     `json:"name"`
+	AllowFailure bool       `json:"allow_failure"`
+	Author       Author     `json:"author"`
+	Description  *string    `json:"description"`
+	Sha          string     `json:"sha"`
+	TargetURL    string     `json:"target_url"`
+	FinishedAt   *time.Time `json:"finished_at"`
+	ID           int        `json:"id"`
+	Ref          string     `json:"ref"`
+}
+
+type Author struct {
```

Comment:
I think I just missed the existing `User` struct since the name wasn't a match to the JSON key. In order to support all the fields the commit status API returns we would need to add a new field to handle `web_url` in the response. I'll take a pass at that.